src/Utils.cc

```diff
@@ -362,5 +362,87 @@ void copyChildren(ElementPtr _sdf, tinyxml2::XMLElement *_xml,
     }
   }
 }
+
+/////////////////////////////////////////////////
+std::string readXMLStream(std::istream &_in)
```

Comment:
Would it be possible to use `tinyXML` to do this instead of manually parsing the stream? This seems brittle and I'm not sure if it handles escapes and `cdata` content.